solidity/contracts/PetalLock.sol

```diff
@@ -108,16 +110,20 @@ contract PetalLock {
 	}
 
 	function makeImmutable(address originalOwner, SubDomainLabelNode[] memory pathToChild, bytes memory contenthash, address resolutionAddress) private {
+		uint256 pathToChildLength = pathToChild.length;
+		require(pathToChildLength > 0, 'PetalLock: Missing subdomain path');
 		require(ensRegistry.recordExists(pathToChild[0].node), 'PetalLock: The second level domain does not exist');
 		require(ensTokenWrapper.isWrapped(pathToChild[0].node), 'PetalLock: The second level domain is not wrapped');
-		uint256 finalChildIndex = pathToChild.length - 1;
+
+		uint256 finalChildIndex;
+		unchecked { finalChildIndex = pathToChildLength - 1; }
```

Comment:
I don't think the gas benefits are worth it for this.  The only time I think `unchecked` blocks are worth the gas benefits are when you have it in a tight loop.  In this case, it is called once so we are talking about a single or low double digit gas savings, but comes with a high auditability cost.

Reply:
Same comment for anywhere else `unchecked` is used.

Reply:
removed them. Yeah, they are all one time things